frontend_tests/node_tests/subs.js

```diff
@@ -1,191 +1,215 @@
-global.$ = require('jquery')(window);
-set_global('window', window);
-zrequire('bootstrap', 'third/bootstrap/js/bootstrap');
-
-subs.stream_name_match_stream_ids = [];
-subs.stream_description_match_stream_ids = [];
+set_global('$', global.make_zjquery());
 
 run_test('filter_table', () => {
-    var denmark = {
+    var stream_list = $(".streams-list");
+
+    var scrolltop_called = false;
+    stream_list.scrollTop = function (set) {
+        scrolltop_called = true;
+        if (!set) {
+            return 10;
+        }
+        assert.equal(set, 10);
+    };
+
+    // set-up sub rows stubs
+    var sub_row_data = {};
+    sub_row_data[1] = {
+        elem: 'denmark',
         subscribed: false,
         name: 'Denmark',
         stream_id: 1,
         description: 'Copenhagen',
     };
-    var poland = {
+    sub_row_data[2] = {
+        elem: 'poland',
         subscribed: true,
         name: 'Poland',
         stream_id: 2,
         description: 'monday',
     };
-    var pomona = {
+    sub_row_data[3] = {
+        elem: 'pomona',
         subscribed: true,
         name: 'Pomona',
         stream_id: 3,
         description: 'college',
     };
-    var cpp = {
+    sub_row_data[4] = {
+        elem: 'cpp',
         subscribed: true,
         name: 'C++',
         stream_id: 4,
+        description: 'programming lang',
     };
 
-    var elem_1 = $(global.render_template("subscription", denmark));
-    var elem_2 = $(global.render_template("subscription", poland));
-    var elem_3 = $(global.render_template("subscription", pomona));
-    var elem_4 = $(global.render_template("subscription", cpp));
+    var sub_stubs = [];
+    _.each(sub_row_data, function (data) {
+        var sub_row = ".stream-row-" + data.elem;
+        sub_stubs.push(sub_row);
+
+        $(sub_row).attr("data-stream-id", data.stream_id);
+        $(sub_row).set_find_results('.sub-info-box [class$="-bar"] [class$="-count"]', $(".tooltip"));
+        $(sub_row).detach = function () {
+            return sub_row;
+        };
+    });
+
+    var tooltip_called = false;
+    $(".tooltip").tooltip = function (obj) {
+        tooltip_called = true;
+        assert.deepEqual(obj, {
+            placement: 'left',
+            animation: false,
+        });
+    };
 
-    $("body").empty();
-    $("body").append('<div id="subscriptions_table"></div>');
-    var streams_list = $('<div class="streams-list"></div>');
-    $("#subscriptions_table").append(streams_list);
+    stream_data.get_sub_by_id = function (id) {
+        return sub_row_data[id];
+    };
 
-    stream_data.add_sub("Denmark", denmark);
-    stream_data.add_sub("Poland", poland);
-    stream_data.add_sub("Pomona", pomona);
-    stream_data.add_sub("C++", cpp);
+    $.stub_selector("#subscriptions_table .stream-row", sub_stubs);
+
+    var sub_table = $('#subscriptions_table .streams-list');
+    var sub_table_append = [];
+    sub_table.append = function (rows) {
+        if (typeof rows === "string") {
```

Comment:
We should see if dealing with both types is some strange artifact of either zjquery being strange, or the code doing strange things, or whatever.